extensions/ql-vscode/src/common/disposable-object.ts

```diff
@@ -82,3 +82,12 @@ export abstract class DisposableObject implements Disposable {
     }
   }
 }
+
+export class BasicDisposableObject extends DisposableObject {
+  constructor(...dispoables: Disposable[]) {
```

Comment:
(Feel free to do now, ignore, or leave for a followup PR)

Could this just be the constructor of `DisposableObject`, and avoid adding a new class? The case of passing no args is still allowed and would be a no-op, so it seems like it would be backwards compatible with existing uses of `DisposableObject`.

Reply:
`DisposableObject` is currently abstract, so I'd need to change that, too. But maybe that's not a terrible thing.